src/Kunstmaan/MediaBundle/Validator/Constraints/HasGuessableExtensionValidator.php

```diff
@@ -55,11 +66,30 @@ public function validate($value, Constraint $constraint)
         }
     }
 
+    private function guessMimeType($pathName)
+    {
+        if ($this->mimeTypeGuesser !== null) {
+            return $this->mimeTypeGuesser->guess($pathName);
+        }
+
+        return $this->mimeTypes->guessMimeType($pathName);
+    }
+
+    private function getExtensions($mimeType)
+    {
+        if ($this->extensionGuesser !== null) {
+            return $this->extensionGuesser->guess($mimeType);
+        }
+
+        return $this->mimeTypes->getExtensions($mimeType)[0] ?? null;
+    }
+
     /**
      * @param ExtensionGuesserFactoryInterface $extensionGuesserFactory
      */
     public function setExtensionGuesser(ExtensionGuesserFactoryInterface $extensionGuesserFactory)
     {
+        @trigger_error('Calling the setExtensionGuesser method of this service is deprecated since KunstmaanMediaBundle 5.6 and will be replaced by the "@mime.types" service in KunstmaanMediaBundle 6.0.', E_USER_DEPRECATED);
```

Comment:
```suggestion
        @trigger_error('Calling the "setExtensionGuesser" method on "\Kunstmaan\MediaBundle\Validator\Constraints\ HasGuessableExtensionValidator" is deprecated since KunstmaanMediaBundle 5.6 and the method will be removed in KunstmaanMediaBundle 6.0. Inject the required services through the constructor instead.', E_USER_DEPRECATED);
```